jaraco/functools.py

```diff
@@ -380,7 +488,11 @@ def retry_call(func, cleanup=lambda: None, retries=0, trap=()):
     return func()
 
 
-def retry(*r_args, **r_kwargs):
+def retry(
+    cleanup: Callable[..., None] = lambda: None,
+    retries: int | float = 0,
+    trap: type[BaseException] | tuple[type[BaseException], ...] = (),
+) -> Callable[[Callable[..., R]], Callable[..., R]]:
```

Comment:
I don't care for how type annotations are forcing the signature to be repeated over and over. Is there a way to re-use the signature and avoid the repetition (reflect the remaining args to `retry_call`)?

Reply:
I guess the best idea is to repeat the signature in the stub file and bring back the variadic signature in the implementation.

Reply:
The variadic signature has been restored.